app/src/main/java/nl/tudelft/cs4160/trustchain_android/peersummary/PeerSummaryActivity.java

```diff
@@ -468,85 +278,136 @@ public void onClick(DialogInterface dialog, int id) {
     }
 
     /**
-     * Toggle developer options to connect manually to an ip.
-     * @param buttonView
-     * @param isChecked
+     * sign a received halfblock and directly send this block back to the peer.
+     * @param linkedBlock
      */
-    @Override
-    public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
-        developerMode = isChecked;
-        if (isChecked) {
-            extraInformationPanel.setVisibility(View.VISIBLE);
-            developerModeText.setTextColor(getResources().getColor(R.color.colorAccent));
-        } else {
-            extraInformationPanel.setVisibility(View.GONE);
-            developerModeText.setTextColor(getResources().getColor(R.color.colorGray));
-        }
-    }
+    public void signAndSendHalfBlock(MessageProto.TrustChainBlock linkedBlock) {
+        DualSecret keyPair = Key.loadKeys(this);
+        MessageProto.TrustChainBlock block = createBlock(null, DBHelper,
+                keyPair.getPublicKeyPair().toBytes(),
+                linkedBlock, inboxItemOtherPeer.getPublicKeyPair().toBytes());
+
+        final MessageProto.TrustChainBlock signedBlock = sign(block, keyPair.getSigningKey());
+
+        //todo again we could do validation?
+        DBHelper.insertInDB(signedBlock); // See read the docs (should be signed though)
 
+        new Thread(new Runnable() {
```

Comment:
We might want to add a function with a callback or lambda, which starts a thread and sends a block/crawl message